src/legacy/index.ts

```diff
@@ -7,6 +7,8 @@ import objectHash = require('object-hash');
 
 export { depTreeToGraph, graphToDepTree, DepTree };
 
+const CYCLE_PRUNED_NODE_ID = 'cycle-pruned';
```

Comment:
This is not a node id, a label key, and probably not worth having as a const. Please inline (and also see below regarding label)